File: src/transformers/features/http/HttpAssertQueryTransformer.ts
import { HttpAssertQueryProgrammer } from "../../../programmers/http/HttpAssertQueryProgrammer";

import type { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace HttpAssertQueryTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "http.assertQuery",
      write: HttpAssertQueryProgrammer.write,
    });
}

File: packages/typia/native/transform/features/http/HttpAssertQueryTransformer.go
package http

import (
  shimast "github.com/microsoft/typescript-go/shim/ast"
  nativecontext "github.com/samchon/typia/packages/typia/native/core/context"
  nativehttpprogrammers "github.com/samchon/typia/packages/typia/native/core/programmers/http"
  nativeinternal "github.com/samchon/typia/packages/typia/native/transform/internal"
)

type httpAssertQueryTransformerNamespace struct{}

var HttpAssertQueryTransformer = httpAssertQueryTransformerNamespace{}

func (httpAssertQueryTransformerNamespace) Transform(props nativeinternal.ITransformProps) *shimast.Node {
  return nativeinternal.GenericTransformer.Scalar(nativeinternal.GenericTransformer_IProps{
    ITransformProps: props,
    Method:          "http.assertQuery",
    Write: func(x nativecontext.IProgrammerProps) *shimast.Node {
      return nativehttpprogrammers.HttpAssertQueryProgrammer.Write(nativehttpprogrammers.HttpAssertQueryProgrammer_IProps{
        Context: x.Context,
        Modulo:  x.Modulo,
        Type:    x.Type,
        Name:    x.Name,
        Init:    x.Init,
      })
    },
  })
}
